### ISSUE TITLE
Files generated by Rmd not copied to outputs

### ISSUE THREAD

**Issue by thibautjombart** [reconhub/reportfactory#74]
Please place an "x" in all the boxes that apply
---------------------------------------------

 - [x] I have the most recent version of reportfactory and R
 - [x] I have found a bug
 - [ ] I have a [reproducible example](http://reprex.tidyverse.org/articles/reprex-dos-and-donts.html)
 - [ ] I want to request a new feature

--------

Files generated by a Rmd are not currently copied to the output folder, except for the main output (html by default). For instance, if one sets the following option to produce figures in a *figures/* folder:

```r

```{r setup, include=FALSE}
knitr::opts_chunk$set(
                      fig.path = "figures/")
```

Figures will be produced in  *report_sources/figures/* but not copied over.

**Comment by tjtnew** [comment]
Cheers.  So I think we resolve this by snapshotting the sources folder before compiling reports and then again afterwards.  The different files then get copied over.  This does then render the current setup a little redundant so want to think about what the benefit is of combining the two approaches as opposed to just using the solution to this bug for everything.

**Comment by thibautjombart** [comment]
This was the old solution, aye, but it does pose the problem that if (spurious) files do exist prior to compilation, but are re-generated by the compilation, they will be overlooked when copying things over. Hence the cleaning procedure which removed non-Rmd files from report_sources/ in the old version, but this is a bit aggressive an option. 

Maybe worth considering using  `file.info(...)` or `file.mtime(...)` to get the list of files modified after compilation started, and define these as files to move? This means `compile_reports()` will not be parallelizable, but not a major hindrance I would think.

Thoughts?

**Comment by tjtnew** [comment]
Maybe but I'd be wary of using any file meta information in case something funny happens if people used synced folders etc etc.  We could just copy the reports to the `outputs` folder and then compile in there.  This would only be problematic if people referred to other data within `report_sources` but we can advise them not to do this.  Relative links would work as `outputs` is at the same folder level as `report_sources`

**Comment by tjtnew** [comment]
closed by #76
